### DISCUSSION TITLE
Dynamic commands as rendered output?

### DISCUSSION THREAD

**Discussion by arrpii** [SilentVoid13/Templater#1168] [Help] [answered]
I'm trying to use dynamic commands to call a user script that returns a markdown string.
But my output is not being rendered, I only get the text.
Instead of:
### Header
- Item a
- Item b

I get:
`### Header - Item a - Item b`

I'm trying to build a dynamic summery at the top of a page and I like how minimal the <%+ tp.user.summery(tp) %> command is.
Is there a way of getting rendered markdown from a dynamic command?

**Comment by Zachatoo** [comment]
I don't know how you're adding content to your note, but you should use `tR` to add content to your note from an execution command.

In your MD file:
```js
<%* tp.file.summery(tp, tR) %>
```

In your user script:
```js
function summery(tp, tR) {
  // do whatever you need to do with tp

  // use tR to add content
  tR += "##Header\n\n- Item a\n- Item b";
}
module.exports = summery;
```

Alternatively, you could do something like this, without using tR or an execution command:

```js
<% tp.file.summery(tp) %>
// OR
<%* tR += tp.file.summery(tp) %>
```

```js
function summery(tp, tR) {
  // do whatever you need to do with tp

  // return content to be added to note, either in a non-execution command or in an execution command using tR
  return "##Header\n\n- Item a\n- Item b";
}
module.exports = summery;
```

**Reply by Zachatoo** [reply, marked as answer]
I just realized you were working with dynamic commands, not execution commands, apologies. I have no idea. Dynamic commands are fairly buggy and will likely be deprecated #913

I highly recommend using Dataview for dynamic content instead.

**Reply by arrpii** [reply]
Thanks for the thorough answer @Zachatoo. I wasn't aware that dynamic commands are likely to be deprecated, I'll check out dataview for this purpose.
